### ISSUE TITLE
[11] Fields with invisible attribute left huge blank space

### ISSUE THREAD

**Issue by guidev224** [odoo/odoo#28370]
Impacted versions: 11

Steps to reproduce: Inherit a view and give some fields invisible attribute set to 1 or add new fields with same logic

Current behavior: Blank space left for invisible fields
```
group element rendering (somehow)
 <table>
            <tr>(height=4 according to browser)
                <td>(height=4 according to browser)
                    <label/> (Field's label : Element with display: none in css)
                </td>
                <td>(height=4 according to browser)
                    <div/> (Field's input : Element with display: none in css)
                </td>
            </tr>
        </table>
```

Expected behavior:
Not leave blank spaces for fields with invisible attribute
Video/Screenshot link (optional):

![image](https://user-images.githubusercontent.com/31063466/47945334-19732680-def9-11e8-9925-587b2b738637.png)

**Comment by pedrobaeza** [comment]
The problem is you are not using proper view elements. You shouldn't abuse table, div, and so on. Just `group` and `field` elements.

Closing as no real issue, but misuse of the views possibilities.

**Comment by guidev224** [comment]
@pedrobaeza you didn't get my point.
The code snippet in my code above is not what I added in my xml view, it is how odoo renders `group `element containing fields with `invisible `attribute(from what I've understood). It was just to illustrate what I thought is the cause of the problem.
This is a snippet of code from account_invoice_view.xml (account module in odoo's addons)
```
<group>
    <group>
          ...
    </group>
    <group>
          ...
          <field name="date_invoice" string="Bill Date"/>
          ...
    </group>
</group>
```
The 3 points mean there is at least one field at that place.
Now here is what I have done in my inheriting view

```
<field name="date_invoice" position="after">
      <field name="x_acte_def_sans_paiement" invisible="1"/>
      <field name="x_reservation_paiement_total" invisible="1"/>
       <field name="x_date_abandon_option" attrs="{'invisible':[('state','!=', 'option_abandonnee')]}"
                           readonly="1"/>
       <field name="raison_abondonner" attrs="{'invisible':[('state','!=', 'option_abandonnee')]}"
                           readonly="1"/>
</field>
```
When the value of the `invisible `attribute is `1 `or `True`, only the `label `and the `div `that contains the field's input are set to `display none`.Their parent `td `and their parent's parent `tr `have a` height = 4`, in the browser.
I hope it is clear now.

**Comment by tanghulu0608** [comment]
I have the same issue.

**Comment by guidev224** [comment]
I have managed to remove blank spaces by overriding the following css:
`.o_form_view .o_group .o_td_label + td {
    padding: 2px 36px 2px 8px;
}
`
Just set the padding-top and padding-bottom to 0px
`.o_form_view .o_group .o_td_label + td {
    padding: 0px 36px 0px 8px;
}
`

**Comment by qsm-odoo** [comment]
Fixed in 11.0 with https://github.com/odoo/odoo/commit/886967766965f7781d8b7c5d87d8505bb6881747
